creator-node/src/middlewares.js

```diff
@@ -144,8 +149,13 @@ async function getOwnEndpoint (req) {
   return spInfo['endpoint']
 }
 
-/** Get all creator node endpoints for user by wallet from discprov. */
-async function getCreatorNodeEndpoints (req, wallet) {
+/**
+ * Get all creator node endpoints for user by wallet from discprov
+ *
+ * If tx blocknumber provided, poll discprov until it has indexed that blocknumber
```

Comment:
Let's update the comment to include the max timeout / retries (which exist) I was a bit thrown off by this comment but functionality looks great

Reply:
also a description of the args might be useful? i found `ownEndpoint` a bit confusing

Reply:
i wonder if `ownEndpoint = null, ensurePrimary = false` or default args might make sense here?

Reply:
agreed on comments will do. for defaults - i chose not to in favor of explicitness. its only called twice i figured no real benefit for defaulting

Reply:
+1 for comments

Reply:
added comments, lmk